htlcswitch/failure.go

```diff
@@ -47,203 +46,15 @@ type ErrorDecrypter interface {
-	return s.EncryptError(true, b.Bytes()), nil
-}
-
-// EncryptMalformedError is similar to EncryptFirstHop (it adds the MAC), but
-// it accepts an opaque failure reason rather than a failure message. This
-// method is used when we receive an UpdateFailMalformedHTLC from the remote
-// peer and then need to convert that into an proper error from only the raw
-// bytes.
-//
-// NOTE: Part of the ErrorEncrypter interface.
-func (s *SphinxErrorEncrypter) EncryptMalformedError(reason lnwire.OpaqueReason) lnwire.OpaqueReason {
-	return s.EncryptError(true, reason)
-}
-
-// IntermediateEncrypt wraps an already encrypted opaque reason error in an
-// additional layer of onion encryption. This process repeats until the error
-// arrives at the source of the payment. We re-encrypt the message on the
-// backwards path to ensure that the error is indistinguishable from any other
-// error seen.
-//
-// NOTE: Part of the ErrorEncrypter interface.
-func (s *SphinxErrorEncrypter) IntermediateEncrypt(reason lnwire.OpaqueReason) lnwire.OpaqueReason {
-	return s.EncryptError(false, reason)
-}
-
-// Type returns the identifier for a sphinx error encrypter.
-func (s *SphinxErrorEncrypter) Type() EncrypterType {
-	return EncrypterTypeSphinx
-}
-
-// Encode serializes the error encrypter' ephemeral public key to the provided
-// io.Writer.
-func (s *SphinxErrorEncrypter) Encode(w io.Writer) error {
-	ephemeral := s.EphemeralKey.SerializeCompressed()
-	_, err := w.Write(ephemeral)
-	return err
-}
-
-// Decode reconstructs the error encrypter's ephemeral public key from the
-// provided io.Reader.
-func (s *SphinxErrorEncrypter) Decode(r io.Reader) error {
-	var ephemeral [33]byte
-	if _, err := io.ReadFull(r, ephemeral[:]); err != nil {
-		return err
-	}
-
-	var err error
-	s.EphemeralKey, err = btcec.ParsePubKey(ephemeral[:], btcec.S256())
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// Reextract rederives the error encrypter from the currently held EphemeralKey.
-// This intended to be used shortly after Decode, to fully initialize a
-// SphinxErrorEncrypter.
-func (s *SphinxErrorEncrypter) Reextract(
-	extract ErrorEncrypterExtracter) error {
-
-	obfuscator, failcode := extract(s.EphemeralKey)
-	if failcode != lnwire.CodeNone {
-		// This should never happen, since we already validated that
-		// this obfuscator can be extracted when it was received in the
-		// link.
-		return fmt.Errorf("unable to reconstruct onion "+
-			"obfuscator, got failcode: %d", failcode)
-	}
-
-	sphinxEncrypter, ok := obfuscator.(*SphinxErrorEncrypter)
-	if !ok {
-		return fmt.Errorf("incorrect onion error extracter")
-	}
-
-	// Copy the freshly extracted encrypter.
-	s.OnionErrorEncrypter = sphinxEncrypter.OnionErrorEncrypter
-
-	return nil
-
-}
-
-// A compile time check to ensure SphinxErrorEncrypter implements the
-// ErrorEncrypter interface.
-var _ ErrorEncrypter = (*SphinxErrorEncrypter)(nil)
-
 // OnionErrorDecrypter is the interface that provides onion level error
```

Comment:
not critical to this pr, but is this still the right place for `OnionErrorDecrypter`?

Reply:
yeah, the same thought crossed my mind, but I wanted to do the minimum required for the receiver side.